src/design-system/Context.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { ColorModeProvider } from './color/ColorMode';
+import { ColorMode } from './color/palettes';
+
+type DesignSystemContextValue = {
+  experimentalFlags: {
+    androidShadowsV2?: boolean;
+  };
+};
+
+const DesignSystemContext = React.createContext<DesignSystemContextValue>({
+  experimentalFlags: {
+    androidShadowsV2: false,
+  },
+});
+
+export function DesignSystemProvider({
+  children,
+  colorMode,
+  experimentalFlags = {},
+}: {
+  children: React.ReactNode;
+  colorMode: ColorMode;
+  experimentalFlags?: DesignSystemContextValue['experimentalFlags'];
+}) {
+  return (
+    <DesignSystemContext.Provider value={{ experimentalFlags }}>
```

Comment:
This context value should be memoized.

Reply:
Alternatively, you could have a provider dedicated to experimental flags and then pass it in directly rather than wrapping it in another object. The only downside is that it means consumers have to be careful not to provide a new experimental flags object on every render, but that's probably okay for now given this is still considered private.